# NOTEBOOK RAPPORT SLL

L'objectif de ce notebook est de remplir le rapport du Service du Livre et de la Lecture (SLL).

# A - Site internet

## A1 - Site internet

* Site internet 
* Statut site : Site internet propre

La bibliothèque a-t-elle déposé une attestation de conformité de son site web aux exigences du Référentiel général d'accessibilité pour les administrations RGAA) ? **Non**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style("darkgrid")
import datetime

from kiblib.utils.db import DbConn
from kiblib.utils.code2libelle import Code2Libelle
from kiblib.adherent import Adherent
from kiblib.pret import Pret
from kiblib.document import Document

In [2]:
db_conn = DbConn().create_engine()

year = 2024
#date_extraction_usagers = '2023-12-27'
date_extraction_usagers = '2024-12-25'

# C - Accès et installations
## C1 - Accès et installations

|Nombre d'heures d'ouverture hebdomadaires tous publics|54|C101|
|:-----------------------------------------------------|---|----|
|Nombre de jours d'ouverture hebdomadaires|6|C115|
|Nombre de jours d'ouverture annuels|???|C102|
|La bibliothèque est-elle ouverte dimanche ?|Oui|C116|
|Nombre de places assises disponibles |???|C103 | 
| - Avec possibilité de connexions aux ordinateurs |47|C104|
|Nombre de postes informatiques publics avec accès internet|???|C106|
|La bibliothèque met-elle à disposition des usagers empêchés de lire du fait d’un handicap, des postes informatiques équipés d’aides matérielles et logicielles pour accéder aux collections ?|Non|C114|
|Nombre de photocopieurs en accès public| 1|C109|
|Nombre d'imprimantes en accès public|???|C110
|Nombre de tablettes mises à disposition du public|0|C118|
|Nombre de bibliobus|1|C112
|Liseuses mises à la dispsition du public|0|C118|
|Bornes d'écoute|1|C119|
|Nombre de consoles de jeux-vidéos|6|C121|
|Connexion wifi publique|oui|C120|

## C2 - Catalogue

### Notices

In [3]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm"""
r = pd.read_sql(query, db_conn)
r

COUNT(bm.biblionumber)
0                  236170

### Notices créées dans l'année

In [4]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

COUNT(bm.biblionumber)
0                    8861

### Notices créées dans l'année "à la main"

In [5]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'
    AND ExtractValue( bm.metadata, '//datafield[@tag="801"]/subfield[@code="b"]' ) = '595126101'"""
r = pd.read_sql(query, db_conn)
r

COUNT(bm.biblionumber)
0                     828

### Nombre de notices entrées par importation dans l'année

In [6]:
# Nombre de notices entrées dans l'année - nombre de notices entrées à la main
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'
    AND ExtractValue( bm.metadata, '//datafield[@tag="801"]/subfield[@code="b"]' ) != '595126101'"""
r = pd.read_sql(query, db_conn)
r

COUNT(bm.biblionumber)
0                    8033

## C3 - Locaux
|Intitulé|valeur|code SLL|
|:--|---|---|
|Surface utile nette totale (en m²)|5300|C301|
|Espaces de services Interns ( en m²)|590|C302|
|Espace de service public (en m²)|3350|C303|
|Espace manifestations et animations (en m²)|250|C304|
|Espace de magains|550|C305|
|Caféterias et surfaces détente|140|C306|
|Votre bâtiment est-il accessible au sens de la loi du 11 février 2005 ?|oui|C309|
|Précisez|Monte-personnes et ascenseurs qui permettent à chacun d'accéder à l'ensemble du bâtiment|C310|


# D - Collections

In [7]:
query = f"""SELECT
                    i.itemnumber,
                    i.barcode,
                    i.dateaccessioned,
                    i.price,
                    i.homebranch,
                    i.holdingbranch,
                    i.location,
                    i.ccode,
                    i.itemcallnumber,
                    i.notforloan,
                    i.damaged,
                    DATE(i.damaged_on),
                    i.withdrawn,
                    DATE(i.withdrawn_on),
                    i.itemlost,
                    DATE(i.itemlost_on),
                    i.onloan,
                    i.datelastborrowed,
                    i.biblionumber,
                    b.title as titre,
                    bi.publicationyear,
                    bi.itemtype,
                    i.timestamp
                FROM koha{year}.items i
                JOIN koha{year}.biblioitems bi ON i.biblionumber = bi.biblionumber
                JOIN koha{year}.biblio b ON i.biblionumber = b.biblionumber
                WHERE i.notforloan IN (-1, -2, 0, 2, -4, -3, 5)
                AND i.homebranch != 'MUS'"""
items = pd.read_sql(query, db_conn)

In [8]:
# On instancie la classe Document
# La classe Document permet d'ajouter les infos dont ont a besoin pour le SLL

exemplaires = Document(df=items, db_conn=db_conn)
exemplaires.get_doc_statdb_data()
exemplaires.get_doc_es_data()

In [9]:
exemplaires.df.columns

Index(['itemnumber', 'barcode', 'dateaccessioned', 'price', 'homebranch',
       'holdingbranch', 'location', 'ccode', 'itemcallnumber', 'notforloan',
       'damaged', 'DATE(i.damaged_on)', 'withdrawn', 'DATE(i.withdrawn_on)',
       'itemlost', 'DATE(i.itemlost_on)', 'onloan', 'datelastborrowed',
       'biblionumber', 'titre', 'publicationyear', 'itemtype', 'timestamp',
       'doc_item_id', 'doc_item_code_barre', 'doc_item_date_creation',
       'doc_item_prix', 'doc_item_site_detenteur_code',
       'doc_item_site_rattachement_code', 'doc_item_localisation_code',
       'doc_item_collection_ccode', 'doc_item_cote', 'doc_statut_code',
       'doc_statut_abime_code', 'doc_statut_desherbe_code',
       'doc_statut_perdu_code', 'doc_usage_emprunt_date',
       'doc_usage_date_dernier_pret', 'doc_biblio_id', 'doc_biblio_titre',
       'doc_biblio_annee_publication', 'doc_biblio_support_code',
       'doc_item_date_modif', 'doc_item_date_creation_annee',
       'doc_item_site_detenteur'

In [10]:
exemplaires.df['sll_acq'] = 'non'
exemplaires.df.loc[exemplaires.df['doc_item_date_creation'].astype('str').str[0:4] == '2023', 'sll_acq' ] = 'oui'

In [11]:
exemplaires_sll = exemplaires.df
exemplaires_sll = exemplaires_sll[['sll_acq', 'sll_public', 'sll_acces', 'sll_collection', 'doc_biblio_id', 'doc_item_id']]

In [12]:
exemplaires_sll['sll_public'].value_counts()

adultes    193890
enfants     71315
Name: sll_public, dtype: int64

In [13]:
exemplaires_sll['sll_acq'].value_counts()

non    251428
oui     13777
Name: sll_acq, dtype: int64

In [14]:
exemplaires_sll['sll_acces'].value_counts()

accès direct      138005
accès indirect    127200
Name: sll_acces, dtype: int64

In [15]:
query = f"""SELECT
                    itemnumber,
                    barcode,
                    dateaccessioned,
                    homebranch,
                    location,
                    ccode,
                    biblionumber,
                    itemtype
                FROM statdb.stat_eliminations
                WHERE annee_mise_pilon = '{year}'"""
elim = pd.read_sql(query, db_conn)

In [16]:
eliminations = Document(df=elim, db_conn=db_conn)
eliminations.get_doc_statdb_data()
eliminations.get_doc_es_data()

In [17]:
eliminations_sll = eliminations.df
eliminations_sll = eliminations_sll[['sll_public', 'sll_acces', 'sll_collection', 'doc_biblio_id', 'doc_item_id']]

## D1 - Livres imprimés

In [18]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D1 - Livres imprimés']
data_el = eliminations_sll[eliminations_sll['sll_collection'] == 'D1 - Livres imprimés']

### Fonds
#### UM - Adultes et Enfants

In [19]:
data.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    126209
enfants     67220
Name: doc_item_id, dtype: int64

#### UM en libre-accès - Adultes et Enfants

In [20]:
data_libre_acces = data[data['sll_acces'] == 'accès direct']
data_libre_acces.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    66611
enfants    32628
Name: doc_item_id, dtype: int64

#### TIT en libre accès - Adultes et Enfants

In [21]:
data_libre_acces = data[data['sll_acces'] == 'accès direct']
data_libre_acces.groupby(['sll_public'])['doc_biblio_id'].nunique()

sll_public
adultes    65343
enfants    28945
Name: doc_biblio_id, dtype: int64

### Acquisitions

In [22]:
data[data['sll_acq'] == 'oui'].groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    5421
enfants    5625
Name: doc_item_id, dtype: int64

In [23]:
data[(data['sll_acq'] == 'oui')
     & (data['sll_acces'] == 'accès direct')].groupby(['sll_public'])['doc_biblio_id'].nunique()

sll_public
adultes    4400
enfants    3787
Name: doc_biblio_id, dtype: int64

#### Eliminations

In [24]:
data_el.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    9246
enfants    5183
Name: doc_item_id, dtype: int64

###  TEST LUCAS

In [25]:
data.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    126209
enfants     67220
Name: doc_item_id, dtype: int64

In [26]:
import numpy as np
data_fonds = data.pivot_table(index=['sll_public','sll_acces'],
                             values=['doc_item_id','doc_biblio_id'],
                             aggfunc="nunique",
                             #margins=True,
                             #margins_name='Total'
                )
#data_fonds['total'] = data_fonds['doc_biblio_id'] + data_fonds['doc_item_id']

data_fonds_concat = pd.concat([
    y.append(y.sum().rename((x, 'Sous-Total')))
    for x, y in data_fonds.groupby(level=0)]).append(data_fonds.sum().rename(('Enfants + Adultes','Total')))
data_fonds_concat

/tmp/ipykernel_31061/2752056215.py:11: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  y.append(y.sum().rename((x, 'Sous-Total')))
/tmp/ipykernel_31061/2752056215.py:11: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  y.append(y.sum().rename((x, 'Sous-Total')))
/tmp/ipykernel_31061/2752056215.py:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data_fonds_concat = pd.concat([


doc_biblio_id  doc_item_id
sll_public        sll_acces                                 
adultes           accès direct            65343        66611
                  accès indirect          56214        59598
                  Sous-Total             121557       126209
enfants           accès direct            28945        32628
                  accès indirect          25145        34592
                  Sous-Total              54090        67220
Enfants + Adultes Total                  175647       193429

## D2 - Publications en séries en cours imprmimées
> Demander à Stéphanie

## D3 - Autres documents

###  Documents cartographiques

In [27]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 – Documents cartographiques']

In [28]:
data['doc_item_id'].nunique()

86

In [29]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

0

###  Musique imprimée

In [30]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 – Musique imprimée']

In [31]:
data['doc_item_id'].nunique()

3199

In [32]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

74

In [33]:
data[(data['sll_acces'] == 'accès direct')]['doc_item_id'].nunique()

1126

In [34]:
data[(data['sll_acces'] == 'accès direct')]['doc_biblio_id'].nunique()

1123

In [35]:
data[(data['sll_acces'] == 'accès direct') & (data['sll_acq'] == 'oui')]['doc_biblio_id'].nunique()

73

###  Documents graphiques

In [36]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 - Documents graphiques']

In [37]:
data['doc_item_id'].nunique()

95

In [38]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

1

###  Autres documents

In [39]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 - Autres documents']

In [40]:
data['doc_item_id'].nunique()

129

In [41]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

44

### Eliminations

In [42]:
data_el = eliminations_sll[eliminations_sll['sll_collection'].str[:2] == 'D3']
len(data_el)

29

## D4 - Documents audiovisuels sur support

In [43]:
exemplaires_sll['sll_collection'].value_counts()

D1 - Livres imprimés                                                                  193429
D4 - Documents audiovisuels fonds adultes / Documents sonores : musique                44586
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes                    12433
D1 - Publications en série imprimées                                                    7134
D3 – Musique imprimée                                                                   3199
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés      1641
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants                     1359
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique                  693
D4 - Total documents multimédia sur support                                              217
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés       204
D3 - Autres documents                                                 

In [44]:
d4_coll = exemplaires_sll[exemplaires_sll['sll_collection'].str[:2] == 'D4']['sll_collection'].unique().tolist()

In [45]:
for c in d4_coll:
    print(c)
    print('-------------')
    data = exemplaires_sll[exemplaires_sll['sll_collection'] == c]
    data_el = eliminations_sll[eliminations_sll['sll_collection'] == c]
    print(f"fonds : {len(data)}")
    print(f"acquisitions : {len(data[data['sll_acq'] == 'oui'])}")
    print(f"eliminations : {len(data_el)}")
    
    print('-------------') 

D4 - Documents audiovisuels fonds adultes / Documents sonores : musique
-------------
fonds : 44586
acquisitions : 602
eliminations : 1237
-------------
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique
-------------
fonds : 693
acquisitions : 22
eliminations : 7
-------------
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés
-------------
fonds : 1641
acquisitions : 43
eliminations : 53
-------------
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes
-------------
fonds : 12433
acquisitions : 374
eliminations : 191
-------------
D4 - Total documents multimédia sur support
-------------
fonds : 217
acquisitions : 1
eliminations : 16
-------------
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés
-------------
fonds : 204
acquisitions : 1
eliminations : 4
-------------
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants
-------------
fonds : 1359
acquisitions : 95
eli

In [46]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D1 - Livres imprimés']
data_el = eliminations_sll[eliminations_sll['sll_collection'] == 'D1 - Livres imprimés']

## D6 - Ressources numériques acquises
> Depuis juin 2023 les ressources ne sont plus acquises directement par la médiathèque de Roubaix, mais par la MEL.

### Bouquets de contenus
Pour chaque item, merci de sélectionner le bouquet principal si vous en avez plusieurs
 
|Intitulé|oui/non|Nom de l'offre|Code SLL
|:---|:---:|---|---|
    |Livre|non|   |D522
|Livres enregistrés|non|   |D524
|Dictionnaires, encyclopédies|non|   |D526|
|Musique|non|   |D528
|Vidéo|non|   |D530|
|Presse|non|   |D532|
|Apprentissage en ligne (adultes)|non|   |D534|
|Ludo-éducatif (jeunesse)|non|   |D536|
|Autres bouquets acquis|non|   |D538

### Ressources à l'unité

|Intitulé|oui/non|Nombre d'acquisitions|Nom du fournisseur|Code SLL|
|:---|---|---|---|---|
|Livres|oui|45|Autre|D539-540-541|
|Presse|non|   |   |   |
Applications|non|   |   |   |
Autres ressources|non|   |   |    |

## D7 - Collections adaptées

|intitulé|oui/non|Code SLL|
|:---|---|---|
|La bibliothèque propose-t-elle des collections adaptées ?|oui|D701
|Livres en gros caractères|oui|D708
|Livres lus du commerce|oui|D709
|Braille|non|D702
|Ouvrages en Daisy|non|
|Revues en synthèse vocale|non|D703
|Revues en synthèse vocale|non|D704
|Livres-DVD en langue des signes française (LSF)|non|D705
|Livres tactiles ou en relief|non|D706
|Autres collections adaptées|oui|D707

# E - Usages et usagers de la bibliothèque

## E1 - Usagers

In [47]:
query = f"""SELECT
    date_extraction,
    age as adh_age,
    geo_ville as adh_geo_ville,
    geo_roubaix_iris as adh_geo_rbx_iris_code,
    sexe as adh_sexe,
    inscription_code_carte as adh_inscription_carte_code,
    inscription_code_site as adh_inscription_site_code,
    inscription_attribut as adh_inscription_attribut_action_code,
    inscription_fidelite as adh_inscription_nb_annees_adhesion,
    nb_venues_prets_mediatheque,
    nb_venues_prets_bus
FROM statdb.stat_adherents WHERE date_extraction = '{date_extraction_usagers}'"""
inscrits = pd.read_sql(query, db_conn)

In [48]:
adh = Adherent(df=inscrits, db_conn=db_conn)
adh.get_adherent_statdb_data()
adh.get_adherent_es_data()

In [49]:
adh.df['nb'] = 1
adh.df['adh_age_sll'] = 'Non pertinent'
adh.df.loc[adh.df['adh_age_lib3'] == '0 - 14 ans', 'adh_age_sll'] = '0 - 14 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '15 - 24 ans', 'adh_age_sll'] = '15 - 64 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '25 - 64 ans', 'adh_age_sll'] = '15 - 64 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '65 ans et plus', 'adh_age_sll'] = '65 ans et plus'
adh.df.loc[adh.df['adh_age_lib3'] == 'Inconnu', 'adh_age_sll'] = 'Inconnu'

### Inscrits actifs

In [50]:
data = adh.df

In [51]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans                    54
                F           2772
                M           2516
                O              1
15 - 64 ans                  127
                F           5328
                M           3433
                O              6
65 ans et plus                 6
                F            515
                M            417
Non pertinent   N            302
Name: nb, dtype: int64

In [52]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        5343
15 - 64 ans       8894
65 ans et plus     938
Non pertinent      302
Name: nb, dtype: int64

In [53]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus               133
                F           5843
                M           3850
                O              6
<= 14 ans                     54
                F           2772
                M           2516
                O              1
Non pertinent   N            302
Name: nb, dtype: int64

In [54]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

10705

### dont nouveaux inscrits

In [55]:
data = adh.df[adh.df['adh_inscription_nb_annees_adhesion'] == 0]
len(data)

6451

In [56]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans                    54
                F           1229
                M           1148
                O              1
15 - 64 ans                  127
                F           2205
                M           1408
                O              5
65 ans et plus                 6
                F            110
                M             94
Non pertinent   N             64
Name: nb, dtype: int64

In [57]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        2432
15 - 64 ans       3745
65 ans et plus     210
Non pertinent       64
Name: nb, dtype: int64

In [58]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus               133
                F           2315
                M           1502
                O              5
<= 14 ans                     54
                F           1229
                M           1148
                O              1
Non pertinent   N             64
Name: nb, dtype: int64

In [59]:
data.groupby(['adh_age_lib2'])['nb'].sum()

adh_age_lib2
15 ans et plus    3955
<= 14 ans         2432
Non pertinent       64
Name: nb, dtype: int64

In [60]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

/tmp/ipykernel_31061/3053756961.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()


4493

In [61]:
data['adh_inscription_carte_personnalite'].value_counts()

Personne        6387
Collectivité      64
Name: adh_inscription_carte_personnalite, dtype: int64

### Emprunteurs actifs

In [62]:
adh.df.columns

Index(['date_extraction', 'adh_age', 'adh_geo_ville', 'adh_geo_rbx_iris_code',
       'adh_sexe', 'adh_inscription_carte_code', 'adh_inscription_site_code',
       'adh_inscription_attribut_action_code',
       'adh_inscription_nb_annees_adhesion', 'nb_venues_prets_mediatheque',
       'nb_venues_prets_bus', 'adh_inscription_carte_personnalite_code',
       'adh_age_code', 'adh_age_lib1', 'adh_age_lib2', 'adh_age_lib3',
       'adh_geo_gentilite', 'adh_geo_rbx_iris', 'adh_geo_rbx_quartier',
       'geo_rbx_secteur', 'inscription_carte', 'adh_inscription_carte_type',
       'adh_inscription_carte_gratuite', 'adh_inscription_carte_prix',
       'adh_inscription_carte_personnalite', 'adh_inscription_site',
       'adh_inscription_nb_annees_adhesion_tra',
       'adh_inscription_attribut_action', 'nb', 'adh_age_sll'],
      dtype='object')

In [63]:
adh.df['emprunteur'] = 'non'
adh.df.loc[adh.df['nb_venues_prets_bus'] > 0, 'emprunteur'] = 'oui'
adh.df.loc[adh.df['nb_venues_prets_mediatheque'] > 0, 'emprunteur'] = 'oui'

data = adh.df[adh.df['emprunteur'] == 'oui']
len(data)

9993

In [64]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans                    30
                F           2128
                M           1838
15 - 64 ans                   42
                F           3260
                M           1685
                O              3
65 ans et plus                 3
                F            434
                M            285
Non pertinent   N            285
Name: nb, dtype: int64

In [65]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        3996
15 - 64 ans       4990
65 ans et plus     722
Non pertinent      285
Name: nb, dtype: int64

In [66]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus                45
                F           3694
                M           1970
                O              3
<= 14 ans                     30
                F           2128
                M           1838
Non pertinent   N            285
Name: nb, dtype: int64

In [67]:
data.groupby(['adh_age_lib2'])['nb'].sum()

adh_age_lib2
15 ans et plus    5712
<= 14 ans         3996
Non pertinent      285
Name: nb, dtype: int64

In [68]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

/tmp/ipykernel_31061/3053756961.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()


6677

In [69]:
data['adh_inscription_carte_personnalite'].value_counts()

Personne        9708
Collectivité     285
Name: adh_inscription_carte_personnalite, dtype: int64

### Fréquentation : entrées dans l'établissement

In [70]:
query = f"""SELECT
    SUM(entrees)
FROM statdb.stat_entrees WHERE YEAR(datetime) = '{year}'"""
entrees = pd.read_sql(query, db_conn)
entrees

SUM(entrees)
0      258427.0

## E2 - Prêts

In [88]:
query = f"""SELECT
    issuedate as pret_date,
    date_due as pret_date_retour_prevue,
    returndate as pret_date_retour_effectif,
    renewals as pret_nb_renouvellement,
    branch as pret_site_pret_code,
    age as adh_age,
    sexe as adh_sexe_code,
    ville as adh_geo_ville,
    iris as adh_geo_rbx_iris_code,
    branchcode as adh_inscription_site_code,
    categorycode as adh_inscription_carte_code,
    fidelite as adh_inscription_nb_annees_adhesion,
    itemnumber as doc_item_id,
    homebranch as doc_item_site_detenteur_code,
    location as doc_item_localisation_code,
    ccode as doc_item_collection_ccode,
    itemcallnumber as doc_item_cote,
    itemtype as doc_biblio_support_code,
    publicationyear as doc_biblio_annee_publication,
    biblionumber as doc_biblio_id,
    dateaccessioned as doc_item_date_creation
FROM statdb.stat_issues
WHERE YEAR(issuedate) = '{year}'"""
prts = pd.read_sql(query, db_conn)
len(prts)

364269

In [89]:
c2l = Code2Libelle(db_conn)
c2l.get_val()
c2l = c2l.dict_codes_lib

prets = Pret(df=prts, db_conn=db_conn, c2l=c2l)
prets.get_pret_statdb_data()
prets.get_pret_es_data()

In [90]:
prets_sll = prets.df[['sll_public', 'sll_acces', 'sll_prets_coll', 'sll_collection', 'sll_prets']]
prets_sll['nb'] = 1

/tmp/ipykernel_31061/3051755607.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  prets_sll['nb'] = 1


In [131]:
# Nb de prêts - hors Collectivités
data = prets_sll.pivot_table(index="sll_prets",
                             columns="sll_public",
                             values="nb",
                             aggfunc=sum,
                             margins=True,
                             margins_name='Total')
data

sll_public                            adultes  enfants   Total
sll_prets                                                     
Autres documents                         1047       85    1132
E2 - Documents vidéo                    18106     4073   22179
E2 - Livres                            105793   201203  306996
E2 – Documents sonores : livres          1472      314    1786
E2 – Documents sonores : musique        13305      491   13796
E2 – Publications en série imprimées    12900     5480   18380
Total                                  152623   211646  364269

In [132]:
# Nb de prêts - Collectivités
data_coll = prets_sll[prets_sll['sll_prets_coll']=="Prêt aux collectivités"].pivot_table(index="sll_prets",
                                                                                         #columns="sll_public",
                                                                                         values="nb",
                                                                                         aggfunc=sum,
                                                                                         margins=True,
                                                                                         margins_name='Total').rename(columns={"nb":"Dont Collectivités"})
data_coll

Dont Collectivités
sll_prets                                               
Autres documents                                      84
E2 - Documents vidéo                                  81
E2 - Livres                                        29290
E2 – Documents sonores : livres                       28
E2 – Documents sonores : musique                      24
E2 – Publications en série imprimées                  52
Total                                              29559

In [136]:
data_synthese = data.merge(data_coll,
                           left_on='sll_prets',
                           right_on='sll_prets',
                           how='left'
                          )
data_synthese = data_synthese[["adultes",'enfants','Total','Dont Collectivités']]

In [138]:
lst = ["E2 - Livres",
       "E2 – Publications en série imprimées",
       "E2 – Documents sonores : musique",
       "E2 – Documents sonores : livres",
       "E2 - Documents vidéo",
       "Autres documents",
       "Total"
      ]
data_synthese.loc[lst]

adultes  enfants   Total  \
sll_prets                                                        
E2 - Livres                            105793   201203  306996   
E2 – Publications en série imprimées    12900     5480   18380   
E2 – Documents sonores : musique        13305      491   13796   
E2 – Documents sonores : livres          1472      314    1786   
E2 - Documents vidéo                    18106     4073   22179   
Autres documents                         1047       85    1132   
Total                                  152623   211646  364269   

                                      Dont Collectivités  
sll_prets                                                 
E2 - Livres                                        29290  
E2 – Publications en série imprimées                  52  
E2 – Documents sonores : musique                      24  
E2 – Documents sonores : livres                       28  
E2 - Documents vidéo                                  81  
Autres documents                                      84  
Total                                              29559

> A la dernière ligne **Tota** il faut ajouter pour valeur le **Nb de prêts de livres numériques**. Ce chiffre est à récupérer auprès de Mathilde L.

## E3 - Autres types d'usages

### Réservations

In [85]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_reserves WHERE YEAR(reservedate) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

COUNT(*)
0     43341

## E5 - Sessions internet (connexions + wifi)

In [86]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_webkiosk WHERE YEAR(heure_deb) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

COUNT(*)
0     55961

In [87]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_wifi WHERE YEAR(start_wifi) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

COUNT(*)
0     13988

### E502 - Nombre de visites du site dans la bibliothèque (visite interne)

In [88]:
query = f"""SELECT SUM(visites) AS 'Total visites interne'
FROM statdb.stat_web2 sw 
WHERE YEAR(`date`) = {year}
AND origine = 'interne'"""

pd.read_sql(query,db_conn)

Total visites interne
0                13920.0

### E503 - Nombre de visite du site internet hors bibliothèque (visite externe)

In [89]:
query = f"""SELECT SUM(visites) AS 'Total visites externe'
FROM statdb.stat_web2 sw 
WHERE YEAR(`date`) = {year}
AND origine = 'externe'"""

pd.read_sql(query,db_conn)

Total visites externe
0               175338.0

# H - Action culturelle (nouvelle version)

In [62]:
import pandas as pd

In [63]:
ac = pd.read_excel("/home/kibini/AC_synthese_2023-2024.xlsx")

/home/kibini/miniconda3/envs/kibini/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Data Validation extension is not supported and will be removed
  warn(msg)


In [64]:
ac = ac[ac['Date'].dt.year==2024]

In [65]:
ac.groupby("Nouveau Type d'action").size()

Nouveau Type d'action
Atelier                 56
Atelier d'écriture       6
Atelier informatique     9
Autre                    2
Club lecture             7
Concert                 11
Conférence              25
Diffusion                6
Exposition               3
Jeux                    10
Jeux-vidéo              27
Lecture                  9
Projection               4
Rencontre               11
Spectacle               16
Séance de contes        20
Vente                    2
dtype: int64

In [66]:
# SLL Types d'action
ac.loc[ac["Nouveau Type d'action"]=="Exposition",["sll_type_action"]] = "H4 - Exposition"

ac.loc[ac["Nouveau Type d'action"].isin(['Conférence','Rencontre','Lecture']),['sll_type_action']] = "H4 - Conférences, rencontres, lectures"

ac.loc[ac["Nouveau Type d'action"].isin(['Concert','Projection']),['sll_type_action']] = "H4 - Concerts, projections"

ac.loc[ac["Nouveau Type d'action"]=="Séance de contes",["sll_type_action"]] = "H4 - Séances de contes"

ac.loc[ac["Nouveau Type d'action"].isin(["Club lecture","Atelier d'écriture"]),['sll_type_action']] = "H4 - Clubs de lecteurs, ateliers d'écriture"

#ac.loc[ac["Nouveau Type d'action"].isin(["Journée festive","Salon du livre","Festival"]),['sll-type_action']] = "H4 - Fêtes, salons du livre, festivals" 
#ac.loc[ac["Evénement"].notna(),["sll_type_action"]] = "H4 - Fêtes, salons du livre, festivals"


ac.loc[~ac["sll_type_action"].isin(["H4 - Exposition",
                                    "H4 - Conférences, rencontres, lectures",
                                    "H4 - Concerts, projections",
                                    "H4 - Séances de contes",
                                    "H4 - Clubs de lecteurs, ateliers d'écriture",
                                    #"H4 - Fêtes, salons du livre, festivals"
                                   ]),
       ["sll_type_action"]] = "H4 - Autres"

> La source de donnée utilisée pour réaliser ces stats est amené à évoluer. Il faudra bien penser à finaliser ce script une fois que les noms de colonnes du fichier Excel Action Culturel seront définitivement fixés

In [67]:
# SLL Types de publics

ac.loc[ac["Public"].isin(["Adolescents","Adultes","Familles","Tout public"]),["sll_type_public"]] = "a/ Tout public"
ac.loc[ac['Public'].isin(["Enfants","Petite enfance"]),["sll_type_public"]] = "b/ Enfants"

In [68]:
ac['Nb actions'] = 1

In [69]:
# import du fichier de suivi des actions AECS
sheets2keep = ['Bruno',
               'Céline',
               'Esther',
               'Gwen',
               'Hélène',
               'Inès',
               'Laetitia C',
               'Laetitia D',
               'Mathilde M',
               'Pascale',
               'Zina',
              ]

aecs = pd.read_excel("/home/kibini/formulaire_AECS.xlsx",sheets2keep)
aecs

/home/kibini/miniconda3/envs/kibini/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Data Validation extension is not supported and will be removed
  warn(msg)


{'Bruno':    Date début action      Date fin action          Champ d'action  \
 0         2024-10-04  2023-10-18 00:00:00   dont Livre et lecture   
 1         2024-10-13  2023-05-24 00:00:00               Education   
 2         2024-10-13  2023-05-24 00:00:00               Education   
 3         2024-10-20  2023-05-23 00:00:00               Education   
 4         2024-10-20  2023-05-23 00:00:00               Education   
 5         2024-11-09  2023-11-09 00:00:00               Education   
 6         2024-11-17          17/11/20223   dont Livre et lecture   
 7         2024-12-07  2023-12-15 00:00:00               Education   
 8         2024-12-14  2024-02-25 00:00:00   dont Livre et lecture   
 9         2024-11-08  2024-12-13 00:00:00   dont Livre et lecture   
 10        2024-01-10  2024-02-21 00:00:00   dont Livre et lecture   
 11        2023-12-11  2024-02-02 00:00:00               Education   
 12        2024-02-01  2024-03-29 00:00:00   dont Livre et lecture   
 13        

In [70]:
aecs = pd.concat(aecs)

In [71]:
effectifs_scolaires = pd.read_excel("/home/kibini/formulaire_AECS.xlsx",sheet_name="Effectifs")
effectifs_scolaires

Nom  Total Maternelle  Total Elementaire  Total Ecole
0               Albert Camus             215.0              324.0          539
1              Albert Samain             150.0                0.0          150
2            Alphonse Daudet              24.0                0.0           24
3             Anatole France             112.0              240.0          352
4              Blaise Pascal              57.0              104.0          161
5          Boileau / Pasteur             156.0              295.0          451
6                     Buffon             127.0              219.0          346
7        Charles De Foucauld               NaN                NaN            0
8           Charles Perrault              32.0                NaN           32
9                  Condorcet               NaN              137.0          137
10            Edmond Rostand              43.0               57.0          100
11          Edouard Vaillant              94.0              103.0          197
12              Elsa Triolet              72.0              135.0          207
13            Ernest Legouvé             107.0              228.0          335
14          Ernest Legouvé 2               NaN                NaN            0
15              Ernest Renan              95.0              222.0          317
16           François Villon             121.0              158.0          279
17               George Sand              80.0                NaN           80
18             Henri Carette             136.0                NaN          136
19           Jacques Prévert             118.0                NaN          118
20                 Jean Macé               NaN              231.0          231
21              Jeanne d'Arc             236.0              428.0          664
22               Jules Ferry              89.0              132.0          221
23              Jules Guesde             123.0              200.0          323
24               Jules Verne             198.0                NaN          198
25                 La Cordée              12.0               48.0           60
26                   Lakanal               NaN              130.0          130
27                 Lavoisier               NaN              266.0          266
28              Léo Lagrange              72.0              178.0          250
29             Léon Gambetta              94.0              156.0          250
30              Léon Jouhaux               NaN              113.0          113
31               Léon Marlot               NaN              223.0          223
32                     Linné             112.0                NaN          112
33                    Littré               NaN              151.0          151
34              Lucie Aubrac             107.0              168.0          275
35        Marie Auxiliatrice             114.0               90.0          200
36               Mère Térèsa             114.0              177.0          291
37                  Michelet             111.0              189.0          300
38                 Montaigne               NaN              146.0          146
39               Montesquieu               NaN              116.0          116
40     Notre Dame de Lourdes              79.0              102.0          181
41  Paul Bert - Edgar Quinet               NaN                NaN            0
42               Paul Valéry              89.0                NaN           89
43        Pierre Brossolette              80.0              148.0          228
44   Pierre de Ronsard niv.1               NaN               98.0           98
45   Pierre de Ronsard niv.2               NaN              154.0          154
46         Pierre de Roubaix               NaN              149.0          149
47            Saint Augustin               NaN                NaN            0
48                Saint Eloi               NaN                NaN            0
49   Saint François d'Assise             104.0              135.0          

In [72]:
effectifs_scolaires.loc[effectifs_scolaires['Nom']=='Saint Louis']["Total Elementaire"].values[0]

148.0

In [73]:
aecs.columns

Index(['Date début action', 'Date fin action', 'Champ d'action',
       'Type d'action', 'Nom action ou projet', 'Lieu',
       'Espace (si hors les murs et pas dans la structure partenaire',
       'Type de public', 'Niveau si école', 'Type d'équipement', 'Structure',
       'Nom d'équipement', 'Nom du porteur de l'action',
       'Nb de classes touchées si public scolaire', 'Nb enfants touchés',
       'Nb adultes touchés', 'Nb d'accueil ou séance',
       'Total personnes touchées', 'Notes', 'Code de malle', 'Titre de malle',
       'Nom emprunteur', 'Nombre de docs', 'Evenement',
       'Espace (si dans les murs)', 'Partenariat', 'Jauge'],
      dtype='object')

### Calcul des actions collectivités

In [74]:
# Remise à plat du df
aecs = aecs.reset_index()

In [75]:
aecs.drop(labels='level_1',axis='columns',inplace=True)

In [76]:
aecs.rename(columns={'level_0':'nom_feuille'},inplace=True)

In [77]:
# Filtre Malles Collectivités
aecs_coll = aecs[aecs["nom_feuille"]=='Laetitia D']

# Sélection des structures écoles

# Liste des noms d'école
school_names = aecs_coll[aecs_coll["Structure"]=='Ecole']["Nom d'équipement"].unique()

for name in school_names:
    # On définit un aecs qui filtre à partir de l'itération school_name
    aecs_coll_school = aecs_coll[aecs_coll["Nom d'équipement"]==name]
    # On calcule un total par types d'actions malles
    Total_actions_BCD = len(aecs_coll_school[aecs_coll_school['Nom action ou projet']=='BCD'])
    Total_actions_MARM = len(aecs_coll_school[aecs_coll_school['Nom action ou projet']=='MARMOTHEQUE'])
    Total_emprunteurs_distincts = aecs_coll_school['Nom emprunteur'].nunique()
    
    # On insère dans le dataframe de base les colonnes du dataframe par école
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_actions_BCD']] = Total_actions_BCD
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_actions_MARM']] = Total_actions_MARM
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_emprunteurs_distincts']] = Total_emprunteurs_distincts

aecs_coll

nom_feuille    Date début action      Date fin action Champ d'action  \
249   Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
250   Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
251   Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
252   Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
253   Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
...          ...                  ...                  ...            ...   
1933  Laetitia D                  NaT                  NaT            NaN   
1934  Laetitia D                  NaT                  NaT            NaN   
1935  Laetitia D                  NaT                  NaT            NaN   
1936  Laetitia D                  NaT                  NaT            NaN   
1937  Laetitia D                  NaT                  NaT            NaN   

     Type d'action Nom action ou projet           Lieu  \
249            NaN               Malles  Hors les murs   
250            NaN               Malles  Hors les murs   
251            NaN               Malles  Hors les murs   
252            NaN               Malles  Hors les murs   
253            NaN               Malles  Hors les murs   
...            ...                  ...            ...   
1933           NaN               Malles            NaN   
1934           NaN               Malles            NaN   
1935           NaN               Malles            NaN   
1936           NaN               Malles            NaN   
1937           NaN               Malles            NaN   

     Espace (si hors les murs et pas dans la structure partenaire  \
249                                                 NaN             
250                                                 NaN             
251                                                 NaN             
252                                                 NaN             
253                                                 NaN             
...                                                 ...             
1933                                                NaN             
1934                                                NaN             
1935                                                NaN             
1936                                                NaN             
1937                                                NaN             

                   Type de public Niveau si école  ...  \
249   Enfant Scolaires (3-12 ans)              CP  ...   
250   Enfant Scolaires (3-12 ans)              MS  ...   
251   Enfant Scolaires (3-12 ans)              GS  ...   
252   Enfant Scolaires (3-12 ans)              CP  ...   
253   Enfant Scolaires (3-12 ans)              CP  ...   
...                           ...             ...  ...   
1933                          NaN             NaN  ...   
1934                          NaN             NaN  ...   
1935                          NaN             NaN  ...   
1936                          NaN             NaN  ...   
1937                          NaN             NaN  ...   

              Titre de malle          Nom emprunteur Nombre de docs Evenement  \
249   Emotions et sentiments       HUCHETTE Juliette             28       NaN   
250         Romans policiers  DRONNEAU Anne-Chrystel             14       NaN   
251              Livres jeux       DEREUMAUX Camille             19       NaN   
252                  Afrique    DUFOURMONT Angélique             28       NaN   
253               Abécédaire     ZIARNOWSKI Delphine             23       NaN   
...                      ...                     ...            ...       ...   
1933                     NaN                     NaN            NaN       NaN   
1934                     NaN                     NaN            NaN       NaN   
1935                     NaN                     NaN            NaN       NaN   
1936                     NaN                   

In [78]:
aecs_coll_merged = aecs_coll.merge(effectifs_scolaires,left_on="Nom d'équipement",right_on="Nom",how="left")

In [79]:
# Calcule des publics touchés

effectif_moyen_classe = 25


# Si au moins 1 action MARM : Compter l'effectif Total Maternlle
aecs_coll_merged.loc[aecs_coll_merged['Total_actions_MARM']>0,['Nb enfants touchés']] = aecs_coll_merged['Total Maternelle']

# Si au moins 1 action BCD : Compter l'effectif total Ecole
aecs_coll_merged.loc[aecs_coll_merged['Total_actions_BCD']>0,['Nb enfants touchés']] = aecs_coll_merged['Total Ecole']

# Aucune action MARM ou BCD : Estimation = Nb emprunteurs distincts x Moyenne d'une classe
aecs_coll_merged.loc[(aecs_coll_merged['Total_actions_MARM']==0)&
            (aecs_coll_merged['Total_actions_BCD']==0),['Nb enfants touchés']] = aecs_coll_merged['Total_emprunteurs_distincts'] * effectif_moyen_classe

aecs_coll_merged

nom_feuille    Date début action      Date fin action Champ d'action  \
0     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
1     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
2     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
3     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
4     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
...          ...                  ...                  ...            ...   
1684  Laetitia D                  NaT                  NaT            NaN   
1685  Laetitia D                  NaT                  NaT            NaN   
1686  Laetitia D                  NaT                  NaT            NaN   
1687  Laetitia D                  NaT                  NaT            NaN   
1688  Laetitia D                  NaT                  NaT            NaN   

     Type d'action Nom action ou projet           Lieu  \
0              NaN               Malles  Hors les murs   
1              NaN               Malles  Hors les murs   
2              NaN               Malles  Hors les murs   
3              NaN               Malles  Hors les murs   
4              NaN               Malles  Hors les murs   
...            ...                  ...            ...   
1684           NaN               Malles            NaN   
1685           NaN               Malles            NaN   
1686           NaN               Malles            NaN   
1687           NaN               Malles            NaN   
1688           NaN               Malles            NaN   

     Espace (si hors les murs et pas dans la structure partenaire  \
0                                                   NaN             
1                                                   NaN             
2                                                   NaN             
3                                                   NaN             
4                                                   NaN             
...                                                 ...             
1684                                                NaN             
1685                                                NaN             
1686                                                NaN             
1687                                                NaN             
1688                                                NaN             

                   Type de public Niveau si école  ...  \
0     Enfant Scolaires (3-12 ans)              CP  ...   
1     Enfant Scolaires (3-12 ans)              MS  ...   
2     Enfant Scolaires (3-12 ans)              GS  ...   
3     Enfant Scolaires (3-12 ans)              CP  ...   
4     Enfant Scolaires (3-12 ans)              CP  ...   
...                           ...             ...  ...   
1684                          NaN             NaN  ...   
1685                          NaN             NaN  ...   
1686                          NaN             NaN  ...   
1687                          NaN             NaN  ...   
1688                          NaN             NaN  ...   

     Espace (si dans les murs) Partenariat Jauge Total_actions_BCD  \
0                          NaN         NaN   NaN               0.0   
1                          NaN         NaN   NaN               0.0   
2                          NaN         NaN   NaN               0.0   
3                          NaN         NaN   NaN               0.0   
4                          NaN         NaN   NaN               0.0   
...                        ...         ...   ...               ...   
1684                       NaN         NaN   NaN               NaN   
1685                       NaN         NaN   NaN               NaN   
1686                       NaN         NaN   NaN               NaN   
1687                       NaN         NaN   NaN               NaN   
1688                       NaN         NaN   NaN               NaN   

     Total_actio

In [80]:
# Suppression des noms doublons pour les noms d'école
aecs_coll_merged.drop_duplicates(subset="Nom d'équipement",
                                 inplace=True
                                )

In [81]:
# tableau spécial collectivités
aecs_coll_merged

nom_feuille    Date début action      Date fin action Champ d'action  \
0     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
6     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
8     Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
10    Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
12    Laetitia D  2023-09-05 00:00:00  2023-10-17 00:00:00            NaN   
...          ...                  ...                  ...            ...   
1137  Laetitia D  2024-10-16 00:00:00  2025-04-23 00:00:00            NaN   
1138  Laetitia D  2024-10-16 00:00:00  2025-04-23 00:00:00            NaN   
1139  Laetitia D  2024-10-17 00:00:00  2025-06-30 00:00:00            NaN   
1243  Laetitia D  2024-11-07 00:00:00  2024-12-19 00:00:00            NaN   
1495  Laetitia D  2025-01-16 00:00:00  2025-07-16 00:00:00            NaN   

     Type d'action Nom action ou projet           Lieu  \
0              NaN               Malles  Hors les murs   
6              NaN               Malles  Hors les murs   
8              NaN               Malles  Hors les murs   
10             NaN               Malles  Hors les murs   
12             NaN               Malles  Hors les murs   
...            ...                  ...            ...   
1137           NaN          MARMOTHEQUE  Hors les murs   
1138           NaN          MARMOTHEQUE  Hors les murs   
1139           NaN               Malles  Hors les murs   
1243           NaN               Malles  Hors les murs   
1495           NaN          MARMOTHEQUE  Hors les murs   

     Espace (si hors les murs et pas dans la structure partenaire  \
0                                                   NaN             
6                                                   NaN             
8                                                   NaN             
10                                                  NaN             
12                                                  NaN             
...                                                 ...             
1137                                                NaN             
1138                                                NaN             
1139                                                NaN             
1243                                                NaN             
1495                                                NaN             

                   Type de public Niveau si école  ...  \
0     Enfant Scolaires (3-12 ans)              CP  ...   
6     Enfant Scolaires (3-12 ans)             CE2  ...   
8     Enfant Scolaires (3-12 ans)              PS  ...   
10    Enfant Scolaires (3-12 ans)              CP  ...   
12    Enfant Scolaires (3-12 ans)             CE2  ...   
...                           ...             ...  ...   
1137     Petite enfance (0-3 ans)             NaN  ...   
1138     Petite enfance (0-3 ans)             NaN  ...   
1139  Enfant Scolaires (3-12 ans)             NaN  ...   
1243  Enfant Scolaires (3-12 ans)              PS  ...   
1495     Petite enfance (0-3 ans)             NaN  ...   

     Espace (si dans les murs) Partenariat Jauge Total_actions_BCD  \
0                          NaN         NaN   NaN               0.0   
6                          NaN         NaN   NaN               0.0   
8                          NaN         NaN   NaN               0.0   
10                         NaN         NaN   NaN               2.0   
12                         NaN         NaN   NaN               2.0   
...                        ...         ...   ...               ...   
1137                       NaN         NaN   NaN               NaN   
1138                       NaN         NaN   NaN               NaN   
1139                       NaN         NaN   NaN               NaN   
1243                       NaN         NaN   NaN               0.0   
1495                       NaN         NaN   NaN               NaN   

     Total_actio

In [82]:
# Afficher le nombre de personnes touchées par école
aecs_coll_merged[aecs_coll_merged["Structure"]=="Ecole"][["Nom d'équipement","Nb enfants touchés"]]

Nom d'équipement Nb enfants touchés
0                 Jules Guesde              375.0
6                 Léo Lagrange              200.0
8           Pierre Brossolette              150.0
10                Léon Jouhaux              113.0
12                   Lavoisier              266.0
21                   Jean Macé              231.0
30          Marie Auxiliatrice              100.0
34                    Michelet              300.0
44     Pierre de Ronsard niv.2              154.0
45                 Montesquieu              116.0
46       Notre Dame de Lourdes              125.0
49           Boileau / Pasteur              156.0
56           Pierre de Roubaix               50.0
57           Sainte Bernadette              140.0
62         Saint Jean Baptiste              200.0
63            Edouard Vaillant              197.0
68    Paul Bert - Edgar Quinet                0.0
72                      Buffon              127.0
73                Albert Camus              325.0
79                 Jules Ferry              221.0
84             Alphonse Daudet               75.0
85               Léon Gambetta              250.0
89               Blaise Pascal              161.0
94                Ernest Renan               95.0
96               Albert Samain               75.0
97     Saint François d'Assise              225.0
101                Paul Valéry               89.0
103               Elsa Triolet              207.0
107     Simone Veil - Niveau 2              100.0
109     Simone Veil - Niveau 1               50.0
110            François Villon              275.0
113                        NaN                NaN
116               Lucie Aubrac              300.0
124                  Condorcet              137.0
130             Anatole France              112.0
137               Jeanne d'Arc              150.0
142                  La Cordée               50.0
143                    Lakanal              150.0
145             Ernest Legouvé              250.0
152           Ernest Legouvé 2              125.0
157                      Linné              112.0
158                     Littré              275.0
166                Léon Marlot              175.0
169           Charles Perrault               25.0
170             Edmond Rostand              100.0
173               Sainte Marie              150.0
178                Jules Verne              125.0
180              Henri Carette              136.0
188                George Sand               80.0
192    Pierre de Ronsard niv.1                NaN
195                Mère Térèsa              291.0
208                  Montaigne              146.0
210                   Voltaire              326.0
981            Jacques Prévert               25.0
1243               Saint Louis               25.0

In [83]:
nb_actions_prets_malles_2024 = aecs_coll_merged['Nb enfants touchés'].sum()

In [84]:
nb_actions_prets_malles_2024

8413.0

### Fin du calcul

In [85]:
# Rajout d'une colonne pour compter le nombre de valeurs
aecs['nb'] = 1

In [86]:
# Si le champ Nb de séance n'est pas rempli, on considère qu'il s'agit d'une occurence unique
aecs.loc[aecs["Nb d'accueil ou séance"].isna(),["Nb d'accueil ou séance"]] = 1

In [87]:
aecs["Total personnes touchées"] = aecs["Nb enfants touchés"] + aecs["Nb adultes touchés"]

## H1 - Partenariats avec des institutions

In [97]:
aecs.dtypes

nom_feuille                                                                 object
Date début action                                                           object
Date fin action                                                             object
Champ d'action                                                              object
Type d'action                                                               object
Nom action ou projet                                                        object
Lieu                                                                        object
Espace (si hors les murs et pas dans la structure partenaire                object
Type de public                                                              object
Niveau si école                                                             object
Type d'équipement                                                           object
Structure                                                                   object
Nom 

In [100]:
aecs.groupby("Structure").size()

Structure
Association                         23
Centre_culturel                      3
Centre_de_loisirs                    5
Centre_social                       30
Collège                              9
Ecole                             1739
Equipement_médicosocial             10
Lycée                                5
Maison_de_retraite                   4
Service_emploi_et_formation          8
Structure_de_la_petite_enfance      69
Supérieur                            7
dtype: int64

In [113]:
# Ajout d'une colonne H1

aecs.loc[aecs['Structure']=="Ecole",
         ["H1 - Partenariats avec des institutions"]] = "a/ Écoles"

aecs.loc[aecs['Structure']=="Collège",
         ["H1 - Partenariats avec des institutions"]] = "b/ Collège"

aecs.loc[aecs['Structure']=="Lycée",
         ["H1 - Partenariats avec des institutions"]] = "c/ Lycée"

aecs.loc[aecs['Structure']=="Supérieur",
         ["H1 - Partenariats avec des institutions"]] = "d/ Supérieur"


In [126]:
# Il n'y a pour l'instant aucun partenariat avec
# Des hôpitaux 
# Des prisons

aecs.loc[aecs['Structure']=="Maison_de_retraite",
         ["H1 - Partenariats avec des institutions"]] = "g/ Maisons de retraite"


aecs.loc[aecs['Structure']=="Centre_social",
         ["H1 - Partenariats avec des institutions"]] = "h/ Centres sociaux"


aecs.loc[aecs['Structure']=="Centre_de_loisirs",
         ["H1 - Partenariats avec des institutions"]] = "i/ Centres de loisirs"

aecs.loc[aecs['Structure']=="Structure_de_la_petite_enfance",
         ["H1 - Partenariats avec des institutions"]] = "j/ Services de la petite enfance"

aecs.loc[aecs['Structure']=="Service_emploi_et_formation",
         ["H1 - Partenariats avec des institutions"]] = "k/ Services de l'emploi"

aecs.loc[aecs['Structure']=="Equipement_médicosocial",
         ["H1 - Partenariats avec des institutions"]] = "l/ Equipement_médicosocial"

In [127]:
aecs[aecs["nom_feuille"]!="Laetitia D"].pivot_table(index='H1 - Partenariats avec des institutions',
                                                    values=["Nb de classes touchées si public scolaire",
                                                            "Nb d'accueil ou séance",
                                                            "Total personnes touchées"],
                                                    aggfunc=sum)[["Nb de classes touchées si public scolaire",
                                                                  "Nb d'accueil ou séance",
                                                                  "Total personnes touchées"
                                                                 ]]

Nb de classes touchées si public scolaire  \
H1 - Partenariats avec des institutions                                              
a/ Écoles                                                                    170.0   
b/ Collège                                                                    31.0   
c/ Lycée                                                                       4.0   
d/ Supérieur                                                                   0.0   
g/ Maisons de retraite                                                         0.0   
h/ Centres sociaux                                                             0.0   
i/ Centres de loisirs                                                          0.0   
j/ Services de la petite enfance                                               0.0   
k/ Services de l'emploi                                                        0.0   
l/ Equipement_médicosocial                                                     0.0   

                                         Nb d'accueil ou séance  \
H1 - Partenariats avec des institutions                           
a/ Écoles                                                 213.0   
b/ Collège                                                 31.0   
c/ Lycée                                                    6.0   
d/ Supérieur                                                7.0   
g/ Maisons de retraite                                      4.0   
h/ Centres sociaux                                         31.0   
i/ Centres de loisirs                                       1.0   
j/ Services de la petite enfance                            8.0   
k/ Services de l'emploi                                    14.0   
l/ Equipement_médicosocial                                  7.0   

                                         Total personnes touchées  
H1 - Partenariats avec des institutions                            
a/ Écoles                                                  3474.0  
b/ Collège                                                  671.0  
c/ Lycée                                                     88.0  
d/ Supérieur                                                209.0  
g/ Maisons de retraite                                       40.0  
h/ Centres sociaux                                          281.0  
i/ Centres de loisirs                                        12.0  
j/ Services de la petite enfance                             15.0  
k/ Services de l'emploi                                      93.0  
l/ Equipement_médicosocial                                   40.0

> Pour la section H1 - Partenariats avec les institutions, on exclus les actions de prêts aux Collectivités

## H2 - Partenariats avec des equipements culturels 

## H3 - Partenariats avec des structures associatives

In [131]:
aecs[aecs["Structure"]=="Association"].groupby("Champ d'action").size()

Champ d'action
Culture    18
dtype: int64

## H4 - Actions au sein de l'établissement

In [88]:
(ac.pivot_table(index='sll_type_action',
               columns='sll_type_public',
               values=['Nb actions',,'Nombre de personnes venues'],
              aggfunc=sum,
               margins=True,
               fill_value=0,
               margins_name='Total')
 .astype(int))

Nb actions                   \
sll_type_public                             a/ Tout public b/ Enfants Total   
sll_type_action                                                               
H4 - Autres                                            113         15   122   
H4 - Clubs de lecteurs, ateliers d'écriture             12          1    12   
H4 - Concerts, projections                              10          5    14   
H4 - Conférences, rencontres, lectures                  43          2    44   
H4 - Exposition                                          3          0     3   
H4 - Séances de contes                                   4         16    19   
Total                                                  176         38   214   

                                            Nombre de personnes venues  \
sll_type_public                                         a/ Tout public   
sll_type_action                                                          
H4 - Autres                                                       4831   
H4 - Clubs de lecteurs, ateliers d'écriture                        130   
H4 - Concerts, projections                                         402   
H4 - Conférences, rencontres, lectures                             911   
H4 - Exposition                                                     65   
H4 - Séances de contes                                             766   
Total                                                             7105   

                                                              
sll_type_public                             b/ Enfants Total  
sll_type_action                                               
H4 - Autres                                        499  5330  
H4 - Clubs de lecteurs, ateliers d'écriture          8   138  
H4 - Concerts, projections                         353   755  
H4 - Conférences, rencontres, lectures              23   934  
H4 - Exposition                                      0    65  
H4 - Séances de contes                             663  1429  
Total                                             1546  8651

## H5 - Actions hors de l'établissement

In [89]:
aecs.groupby("Lieu")[("Nb d'accueil ou séance","Total personnes touchées")].sum()

/tmp/ipykernel_17036/2387532572.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  aecs.groupby("Lieu")[("Nb d'accueil ou séance","Total personnes touchées")].sum()


Nb d'accueil ou séance  Total personnes touchées
Lieu                                                           
Dans les murs                   496.0                    8051.0
Hors les murs                  1734.0                     585.0
Les deux                         70.0                     939.0

In [90]:
nb_actions_hors_les_murs = aecs[aecs["Lieu"]=='Hors les murs']['Total personnes touchées'].sum() + nb_actions_prets_malles_2024

In [91]:
nb_actions_hors_les_murs

8998.0

### Action hors les murs

### Portage à domicile
Le portage à domicile est effectué par M. qui transmet à F. les chiffres chaque année.

## H6 - Actions de coopération

|Intitulé|Oui/Non?|Champ|
|:---|---|---|
|Coopération locale|Oui|H601
|- Dont dont actions de coopération intercommunale (avec des bibliothèques n’appartenant pas à votre collectivité)|Oui|H610|
|Catalogue commun|Oui|H612|
|Portail documentaire commun|Non|H613|
|Politique documentaire commune|Non|H613
|Navette documentaire|Non|H614
|Carte unique de lecteur|Non|H615
|Action culturelle commune|Oui|H617
|Autre action de coopération intercommunal|Non|H616
|Coopération nationale|Oui|H604
|Nombre d'actions|4|H605
|Coopération internationale|Non|H607

## H7 - Actions et services à destination de publics à besoins spécifiques 

In [143]:
aecs[aecs['Structure']=="Association"].pivot_table(index=["Nom d'équipement","Champ d'action"],
                                                   values=["Nb d'accueil ou séance","Total personnes touchées"],
                                                   margins=True,
                                                   aggfunc=sum)

Nb d'accueil ou séance  \
Nom d'équipement                                   Champ d'action                           
ALEFPA                                             Culture                            1.0   
ARA                                                Culture                            1.0   
Accorderie                                         Culture                            5.0   
Amitié Partage                                     Culture                            1.0   
Armée du Salut                                     Culture                            1.0   
CIDFF (Centre d'Information sur les droits des ... Culture                            3.0   
Detournoyment                                      Culture                            1.0   
La Clef                                            Culture                            2.0   
La Maison des Parents                              Culture                            1.0   
Servir                                             Culture                            2.0   
All                                                                                  18.0   

                                                                   Total personnes touchées  
Nom d'équipement                                   Champ d'action                            
ALEFPA                                             Culture                             22.0  
ARA                                                Culture                             15.0  
Accorderie                                         Culture                             45.0  
Amitié Partage                                     Culture                              2.0  
Armée du Salut                                     Culture                             14.0  
CIDFF (Centre d'Information sur les droits des ... Culture                             27.0  
Detournoyment                                      Culture                             30.0  
La Clef                                            Culture                             27.0  
La Maison des Parents                              Culture                              6.0  
Servir                                             Culture                             38.0  
All                                                                                   226.0

In [93]:
liste_publics_specifiques = ["Personnes âgées (65 ans et plus)",
                             "Personnes en situation de handicap",
                             "Jeunes (18-25 ans)",
                             "Petite enfance (0-3 ans)",
                             "Personnes en recherche d'emploi",
                             "Personnes en situation d'illettrisme",
                             "Populations allophones",
                             "Populations en situation d'insertion sociale"
                            ]

In [94]:
aecs.loc[aecs["Type de public"]=="Personnes âgées (65 ans et plus)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "a/ Personnes âgées"

aecs.loc[aecs["Type de public"]=="Personnes en situation de handicap",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "b/ Personnes en situation de handicap"

aecs.loc[aecs["Type de public"]=="Jeunes (18-25 ans)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "c/ Jeunes"

aecs.loc[aecs["Type de public"]=="Petite enfance (0-3 ans)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "d/ Petite enfance"

aecs.loc[aecs["Type de public"]=="Personnes en recherche d'emploi",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "e/ Personnes en recherche d'emploi"

aecs.loc[aecs["Type de public"]=="Personnes en situation d'illettrisme",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "f/ Personnes en situation d'illettrisme"

aecs.loc[aecs["Type de public"]=="Populations allophones",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "g/ Populations non-francophone"

aecs.loc[aecs["Type de public"]=="Populations en situation d'insertion sociale",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "h/ Populations en situation d'insertion sociale"

In [95]:
aecs.pivot_table(index="H7 - Actions et services à destination de publics à besoins spécifiques",
                 values=['nb',"Total personnes touchées"],
                 dropna=False,
                 aggfunc={'nb':sum,"Total personnes touchées":sum}
                )[['nb','Total personnes touchées']]

nb  \
H7 - Actions et services à destination de publi...       
a/ Personnes âgées                                   4   
b/ Personnes en situation de handicap                1   
c/ Jeunes                                           10   
d/ Petite enfance                                   85   
g/ Populations non-francophone                       8   
h/ Populations en situation d'insertion sociale      2   

                                                   Total personnes touchées  
H7 - Actions et services à destination de publi...                           
a/ Personnes âgées                                                       40  
b/ Personnes en situation de handicap                                    16  
c/ Jeunes                                                             197.0  
d/ Petite enfance                                                    1050.0  
g/ Populations non-francophone                                         92.0  
h/ Populations en situation d'insertion sociale                         288

# K - Patrimoine

## K2 - Fonds et acquisitions

In [134]:
query = f"""SELECT
                    i.itemnumber,
                    i.barcode,
                    i.dateaccessioned,
                    i.price,
                    i.homebranch,
                    i.holdingbranch,
                    i.location,
                    i.ccode,
                    i.itemcallnumber,
                    i.notforloan,
                    i.damaged,
                    DATE(i.damaged_on),
                    i.withdrawn,
                    DATE(i.withdrawn_on),
                    i.itemlost,
                    DATE(i.itemlost_on),
                    i.onloan,
                    i.datelastborrowed,
                    i.biblionumber,
                    b.title as titre,
                    bi.publicationyear,
                    bi.itemtype,
                    i.timestamp
                FROM koha{year}.items i
                JOIN koha{year}.biblioitems bi ON i.biblionumber = bi.biblionumber
                JOIN koha{year}.biblio b ON i.biblionumber = b.biblionumber
                WHERE i.notforloan IN (-1, -2, 0, 2, -4, 5)
                AND i.homebranch != 'MUS'"""
items = pd.read_sql(query, db_conn)

In [135]:
exemplaires = Document(df=items, db_conn=db_conn)
exemplaires.get_doc_statdb_data()
exemplaires.get_doc_es_data()

In [136]:
pat_ccode = ['PENACZZ', 'PENCVZZ', 'PENDEZZ', 'PENHPZZ', 'PENPDZZ', 'PENRSZZ', 'AAPPRLP',
             'PPAFIZZ', 'PPEFGZZ', 'PPELGZZ', 'PPEPMZZ', 'PPEPRZZ', 'PPIPIZZ', 'P17']
patrimoine_df = exemplaires.df[exemplaires.df['doc_item_collection_ccode'].isin(pat_ccode)]
patrimoine_df['sll_acq'] = 'non'
patrimoine_df.loc[patrimoine_df['doc_item_date_creation'].astype('str').str[0:4] == '2022', 'sll_acq' ] = 'oui'

/tmp/ipykernel_2550/2479472917.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  patrimoine_df['sll_acq'] = 'non'


In [137]:
len(patrimoine_df)

62528

### Livres imprimés

In [138]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D1 - Livres imprimés']

In [139]:
len(data)

35947

In [140]:
len(data[data['sll_acq'] == 'oui'])

444

In [141]:
data['publicationyear_'] = data['publicationyear'].str.extract(r'(^\d{4})').astype(float)
l1811 = data[data['publicationyear_'] < 1811]
len(l1811)

/tmp/ipykernel_2550/2076920371.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['publicationyear_'] = data['publicationyear'].str.extract(r'(^\d{4})').astype(float)


1055

In [142]:
len(l1811[l1811['sll_acq'] == 'oui'])

33

In [143]:
len(patrimoine_df[patrimoine_df['doc_item_collection_ccode'] == 'AAPPRLP'])

546

In [144]:
len(patrimoine_df[(patrimoine_df['doc_item_collection_ccode'] == 'AAPPRLP') & (patrimoine_df['sll_acq'] == 'oui')])

0

### Publications en séries

In [145]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D1 - Publications en série imprimées']
len(data)

1362

In [146]:
len(data[data['sll_acq'] == 'oui'])

5

### Cartes

In [147]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 – Documents cartographiques']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 0
Acquisitions : 0


### Partitions

In [148]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 – Musique imprimée']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 1661
Acquisitions : 90


### Documents iconographiques

In [149]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 - Documents graphiques']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 94
Acquisitions : 0


### Documents sonores

In [150]:
data = patrimoine_df[patrimoine_df['sll_collection'].isin(['D4 - Documents audiovisuels fonds adultes / Documents sonores : musique',
                                                           'D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés'])]
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 22538
Acquisitions : 157


### Images animées

In [151]:
data = patrimoine_df[patrimoine_df['sll_collection'].isin(['D4 - Documents audiovisuels fonds adultes / documents vidéo adultes',
                                                           'D4 - Documents audiovisuels fonds enfants / documents vidéo enfants'])]
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 895
Acquisitions : 0


D1 - Livres imprimés                                                                  205053
D4 - Documents audiovisuels fonds adultes / Documents sonores : musique                45233
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes                    12067
D1 - Publications en série imprimées                                                    9173
D3 – Musique imprimée                                                                   3193
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants                     1593
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés      1541
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique                  901
D4 - Total documents multimédia sur support                                              287
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés       218
D3 – Documents cartographiques                                                           125
D3 - Documents graphiques                                                                 93
D3 - Autres documents                                                                     83